icicle/src/fields/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+if (EXT_FIELD)
+  set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -DEXT_FIELD")
+endif ()
+
+SET(SUPPORTED_FIELDS_WITHOUT_NTT grumpkin)
+
+set(TARGET icicle_field)
+
+set(SRC ${CMAKE_SOURCE_DIR}/src)
+
+set(FIELD_SOURCE ${SRC}/fields/field.cu)
+list(APPEND FIELD_SOURCE ${SRC}/vec_ops/extern.cu)
+list(APPEND FIELD_SOURCE ${SRC}/fields/mont.cu)
+
+# TODO: impl poseidon for small fields. note that it needs to be defined for the extension field!
+if (DEFINED CURVE)
+  list(APPEND FIELD_SOURCE ${SRC}/poseidon/poseidon.cu)
+  list(APPEND FIELD_SOURCE ${SRC}/poseidon/tree/merkle.cu)
+endif()
+
+if (NOT FIELD IN_LIST SUPPORTED_FIELDS_WITHOUT_NTT)
+  list(APPEND FIELD_SOURCE ${SRC}/ntt/extern.cu)
+  list(APPEND FIELD_SOURCE ${SRC}/ntt/kernel_ntt.cu)
+endif()
```

Comment:
The intention is to do the same for polynomials right?

Reply:
yes